lib/rust/parser/tests/parse.rs

```diff
@@ -75,6 +107,72 @@ fn function_block_simple_args() {
     test("foo a b c =", block![(Function foo #((Ident a) (Ident b) (Ident c)) "=" ())]);
 }
 
+#[test]
+fn code_block_body() {
+    #[rustfmt::skip]
+    let code = [
+        "main =",
+        "    4",
+    ];
+    test(&code.join("\n"), block![(Function main #() "=" (BodyBlock #((Number 4))))]);
+
+    #[rustfmt::skip]
+    let code = [
+        "main =",
+        "    +4",
+        "    print 23",
+    ];
+    #[rustfmt::skip]
+    let expect = block![
+        (Function main #() "=" (BodyBlock #(
+         (OprSectionBoundary (OprApp () (Ok "+") (Number 4)))
+         (App (Ident print) (Number 23)))))
+    ];
+    test(&code.join("\n"), expect);
+}
+
+#[test]
+fn code_block_operator() {
+    let code = ["value = nums", "    * each random", "    + constant"];
+    let expect = block![
+        (Assignment (Ident value) "="
+         (OperatorBlockApplication (Ident nums)
+          #(((Ok "*") (App (Ident each) (Ident random)))
+            ((Ok "+") (Ident constant)))
+          #()))
+    ];
+    test(&code.join("\n"), expect);
+}
+
+#[test]
+fn code_block_argument_list() {
+    #[rustfmt::skip]
+    let code = [
+        "value = foo",
+        "    bar",
+    ];
+    let expect = block![
+        (Assignment (Ident value) "=" (ArgumentBlockApplication (Ident foo) #((Ident bar))))
+    ];
+    test(&code.join("\n"), expect);
+
+
+    #[rustfmt::skip]
+    let code = [
+        "value = foo",
+        "    +1",
+        "    bar",
+    ];
+    #[rustfmt::skip]
+    let expect = block![
+        (Assignment (Ident value) "="
+         (ArgumentBlockApplication (Ident foo) #(
+          (OprSectionBoundary (OprApp () (Ok "+") (Number 1)))
+          (Ident bar))))
+    ];
```

Comment:
Please make sure that the tests cover all possible block types with all possible inonsistencies and errors.

Reply:
I added a lot of weird-indentation test cases; they all parse correctly, but in many cases the whitespace is printed back wrong. I'm looking in to that.

Reply:
All the whitespace problems were caused by an incorrect `impl span::Builder for Option`.